### ISSUE TITLE
[tailer package] macOS bug

### ISSUE THREAD

**Issue by pyed** [fstab/grok_exporter#11]
consider the following code inside a file named `t.go`
``` go
package main

import (
	"fmt"

	"github.com/fstab/grok_exporter/tailer"
)

func main() {
	t := tailer.RunFileTailer("hello", false, nil)

	for line := range t.Lines() {
		fmt.Println(line)
	}
}
```

the file `hello` contains the following
```
line1
```

now run `t.go` it will wait for new lines to be written to `hello`, it won't read anything since we passed `false` to `RunFileTailer`.

now add `line2` to `hello`
``` bash
$ echo "line2" >> hello
```

our program will output the following:
```
line2
```

Great, That's exactly what we're looking for, now let's add `line3`
``` bash
$ echo "line3" >> hello
```

our program will output the following:
```
line1
line2
line3
```
not exactly what we wanted, instead of getting the newest line, we got the whole file.

### env
Go 1.8
macOS 10.12.3
`go test` passes

**Comment by fstab** [comment]
Thanks for the bug report! I fixed it, it should work now. However, I need to get back to it and figure out why this didn't happen in `fileTailer_test`.

**Comment by fstab** [comment]
Hi again. Although my commit fixes the problem, I am still trying to understand why this didn't happen in any tests. One interesting thing I noticed: When you create the file `hello` not in the current directory but in `/tmp/hello`, the bug does not happen. As my tests create temporary logfiles in `/tmp/`, the bug doesn't occur in the tests. I still don't know why this happens, but at least its a starting point to investigate further. I'll comment on this issue when I learned more.

**Comment by fstab** [comment]
Hi, I narrowed it down further: The difference is not if the file is in `/tmp` or not, the real difference is whether the directory is visible in the OS X Finder or not. All files in `/tmp` are not visible in Finder, because of the `com.apple.FinderInfo` attribute on `/private` (`/tmp` is a link to `/private/tmp`). All files starting with a 'dot' are also not visible in Finder.

I suppose the additional `NOTE_ATTRIB` event that caused the issue is triggered by OS X because OS X updates some metadata used for showing the file in Finder. The `grok_exporter` wrongly interpreted this as a truncate event and started reading the file from the beginning. It is fixed now, `NOTE_ATTRIB` events are now ignored unless the file is really truncated.

I created a test that uses a directory in `$HOME` that is not hidden, so this issue is covered by the test now.

Thanks for the bug report!
